frontend/src/components/MapLegend.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+
+interface LegendProps {
+  modeColors: { [key: string]: string };
+}
+
+const MapLegend: React.FC<LegendProps> = ({ modeColors }) => {
+  return (
+    <div className="flex flex-col">
```

Comment:
Can you add `mt-3` here so that the title has some padding above it

Reply:
added